### PR TITLE
add setting and getting container metadata

### LINKED ISSUE #15: Support metadata on containers
```php
$containerClient->setMetdata(["key" => "value"]);
$containerClient->getProperties()->metadata;
```

### PR DESCRIPTION
**Pull request by brecht-vermeersch**

### CODE REVIEW

File: src/Blob/Sas/BlobSasBuilder.php [hunk 1]
```diff
@@ -217,9 +217,9 @@ private function getCanonicalizedResource(string $accountName): string
         return $resource;
     }
 
-    private function dateTo8601Zulu(\DateTime|\DateTimeImmutable $date): string
+    private function dateTo8601Zulu(\DateTimeInterface $date): string
     {
-        return (clone $date)
+        return \DateTime::createFromInterface($date)
```
Comment (brecht-vermeersch) [inline src/Blob/Sas/BlobSasBuilder.php:222]: TIL

File: src/Blob/Models/BlobProperties.php [hunk 2]
```diff
@@ -24,8 +24,13 @@ public function __construct(
 
     public static function fromResponseHeaders(ResponseInterface $response): self
     {
+        $lastModified = \DateTimeImmutable::createFromFormat(\DateTimeInterface::RFC1123, $response->getHeaderLine('Last-Modified'));
+        if ($lastModified === false) {
+            throw new \Exception("Should not happen!");
```
Comment (pimjansen) [inline src/Blob/Models/BlobProperties.php:29]: Lets bring a little better description here since when it does happen an user has no clue where to look

**Review by pimjansen — CHANGES_REQUESTED** [review]

File: src/Blob/Models/BlobProperties.php (hunk 2 above)
Comment (brecht-vermeersch) [reply src/Blob/Models/BlobProperties.php:29]: will fix in next pr

**Review by pimjansen — APPROVED** [review]
LGTM

**Pull request merged**